### PR TITLE
Fix 5 warnings in PhysicsTools/MVATrainer

### PR DESCRIPTION
**Pull request by mrodozov**
#### PR description:
Fixes gcc9 warnings

#### PR validation:

Builds without warnings.

### CODE REVIEW

**Comment by mrodozov** [comment]
please test

**Comment by guitargeek** [comment]
H! Don't you think it would be better to just delete that package? I did some background research on that package some time ago, and it's even older than TMVA. IMHO there is not point in supporting that code and spending time fixing bugs here. Everyone is now either using TMVA or the Python data science ecosystem to train their MVAs.

The only remaining use of that package is within the TopQuarkAnalysis subsystem, where a few tweaks are needed to remove the dependence [1]. It's probably okay to remove the bits of code there that depend on the MVATrainer, as I doubt the TopQuarkAnalysis subsystem is used (seems to date back even to Tevatron times).

What do you think?

[1] https://github.com/cms-sw/cmssw/compare/master...guitargeek:MVATrainer#diff-e90179e3c3ae80c89460b684461cc835

**Comment by mrodozov** [comment]
would you confirm that's the difference 
https://github.com/cms-sw/cmssw/compare/master...guitargeek:MVATrainer ?
because the other link doesn't show any difference (github has its issues) 
Then I can rework the PR

**Comment by guitargeek** [comment]
Yes you are right. Thanks, I updated the link in my post.

Alright then, I will do a PR with the removal of the package and refer to your gcc warnings PR to exemplify how leaving the MVATrainer in the release can be annoying because it raises new warnings for every major gcc version bump. Can you please leave this PR open until the release managers decided if they want to remove the package or not? Do you agree with this procedure?

**Comment by mrodozov** [comment]
> Can you please leave this PR open until the release managers decided if they want to remove the package or not? Do you agree with this procedure?

Yes it fine with me.

File: PhysicsTools/MVATrainer/src/LeastSquares.cc [hunk 1]
```diff
@@ -291,6 +291,7 @@ namespace PhysicsTools {
             break;
           } else
             pos = (Position)(pos + 1);
+          [[fallthrough]];
```
Comment (fabiocos) [inline PhysicsTools/MVATrainer/src/LeastSquares.cc]: @santocch I do not see why this should not be a ```break```, could you please cross check?

**Comment by fabiocos** [comment]
hold

this PR becomes obsolete if #27888 is integrated

**Pull request closed without merging**